src/datapool/datapool_pmem.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * File-backed datapool.
+ * Retains its contents if the pool has been closed correctly.
+ *
+ */
+#include "datapool.h"
+
+#include <cc_mm.h>
+#include <cc_debug.h>
+
+#include <inttypes.h>
+
+#include <libpmem.h>
+
+#define DATAPOOL_SIGNATURE ("PELIKAN") /* 8 bytes */
+#define DATAPOOL_SIGNATURE_LEN (sizeof(DATAPOOL_SIGNATURE))
+#define DATAPOOL_HEADER_LEN 4096
```

Comment:
Do you expect to run Pelikan on architectures that do not support 4k pages? :)

But no, there no correctness (or otherwise) reasons for this number. It's just big enough for any future expansion.

Also, msync() doesn't have to internally align the size if we pass it a nice and round number.

Reply:
got it. we don't plan to use such architectures afaik (though we have evaluated the effect of hugepage before), but this is an open source project so ideally we would like to support a multitude of use cases if we can. 4 KB seems quite large to me but considering the amount of memory we are dealing with using pmem, this seems fine to me.